client/extensions/woocommerce/app/stats/controller.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import page from 'page';
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import { renderWithReduxStore } from 'lib/react-helpers';
+import AsyncLoad from 'components/async-load';
+import StatsPagePlaceholder from 'my-sites/stats/stats-page-placeholder';
+
+function isValidParameters( context ) {
+	const validParameters = {
+		type: [ 'orders', 'customers', 'stock' ],
+		period: [ 'year', 'last_month', 'month', '7day', 'custom' ],
+		segment: [ 'sales_by_date', 'sales_by_product', 'sales_by_category', 'coupon_usage' ]
+	};
+	return Object.keys( validParameters )
+		.every( param => includes( validParameters[ param ], context.params[ param ] ) );
+}
+
+export default function StatsController( context ) {
+	if ( ! isValidParameters( context ) ) {
+		page.redirect( '/store/stats' );
```

Comment:
Fixed, thanks. In the case of invalid slug (domain id), we'll need a page at `/store/stats` as well. Site stats has an aggregate page for all domains, but we could start simply by showing this page

![screen shot 2017-05-01 at 2 04 02 pm](https://cloud.githubusercontent.com/assets/1922453/25570391/f0b5bb22-2e78-11e7-8eb5-10174e1ab5c6.png)

This check will need to access state and likely live in a connected component. I will create a separate issue for this.